src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/executors/spark/SparkExecutor.scala

```diff
@@ -51,9 +52,14 @@ object SparkExecutor extends Executor {
     val id = randomUUID().toString
     val ji = jobInstance.copy(executorJobId = Some(id), jobStatus = Submitting)
     updateJob(ji).map { _ =>
-      val running = getSparkLauncher(id, ji.jobName, ji.jobParameters).launch()
-      Thread.sleep(SparkExecutorConfig.getSubmitTimeOut)
-      running.destroyForcibly()
+      val start = System.currentTimeMillis()
+      val submitTimeOut = SparkExecutorConfig.getSubmitTimeOut
+      val sparkAppHandle = getSparkLauncher(id, ji.jobName, ji.jobParameters).startApplication()
+      while(
+        System.currentTimeMillis() - start <= submitTimeOut &&
+          (sparkAppHandle.getState == SparkAppHandle.State.UNKNOWN || sparkAppHandle.getState == SparkAppHandle.State.CONNECTED)
+      ) Thread.sleep(5.seconds.toMillis)
+      sparkAppHandle.kill()
```

Comment:
Does this not kill the actual spark app and not just the launcher process?

Reply:
Just the launcher process